### DISCUSSION TITLE
Specification version

### DISCUSSION THREAD

**Discussion by CNAChino** [graphql-java/graphql-java#2195] [General]
Hi, was reviewing the release notes and it is noted in 16.1 release that

_"It adds new capabilities based on updated graphql specification..."_

Not sure if this is documented somewhere but just to confirm, may I ask what version of graphql specification is
implemented in 16.1 release and als in 14.1 release.

Thanks in advance!

**Comment by bbakerman** [comment]
We always try to  support the latest published and then key upcoming spec changes that are merged but not official published.

So for example in 16.1 we added support for "multiple directives of the same name" - so called repeatable directives - which is not in a published spec but is merged in the spec repo

https://github.com/graphql/graphql-spec/pull/472

Officially graphql-java 16.x and 14.x support https://spec.graphql.org/June2018/ and elements of the working draft https://spec.graphql.org/draft/

**Comment by crankydillo** [comment]
Sorry if I've missed, but what version is the v21.3 release tied to?  Is that tied to the v2021 spec?

**Reply by dondonz** [reply]
Hello, we always aim to implement the latest draft specification at the time the library is released. Link: https://spec.graphql.org/draft/

The version of the GraphQL specification to read is the latest draft, rather than one of the timestamped versions. I wouldn't use the 2021 version of the spec as quite a bit has changed since then.

Please see this blog post for more explanation: https://www.graphql-java.com/blog/spec-releases-are-not-important
